config/routes.rb

```diff
@@ -30,8 +30,10 @@
 
     patch '/confirm' => 'users/confirmations#confirm'
 
-    match '/edit/email' => 'users/edit_info#email', via: [:get, :put], as: :edit_email
-    match '/edit/mobile' => 'users/edit_info#mobile', via: [:get, :put], as: :edit_mobile
+    get '/edit/email' => 'users/edit_email#edit'
+    match '/edit/email' => 'users/edit_email#update', via: [:patch, :put]
+    get '/edit/mobile' => 'users/edit_mobile#edit'
+    match '/edit/mobile' => 'users/edit_mobile#update', via: [:patch, :put]
```

Comment:
I think these should go in the "non-devise" routes below? IIUC devise_scope is for controllers that inherit from DeviseController, which these do not (see comment above).